pkg/secret/factory.go

```diff
@@ -105,17 +106,24 @@ func (f *Factory) validate() error {
 		}
 	}
 
-	if f.GitUser != "" && !(strings.HasPrefix(f.GitUrl, "http://") || strings.HasPrefix(f.GitUrl, "https://")) {
-		return errors.Errorf("must provide a valid git url for basic auth (ex. https://github.com)")
+	if f.GitUser != "" {
+		if match, _ := regexp.Match(`^(?:https?://)(?:[a-zA-Z0-9\-\.]+(?::\d+)?)(?:\/(?:$|[^/]))?$`, []byte(f.GitUrl)); !match {
+			return errors.Errorf("must provide a valid git url without the repository path for basic auth (ex. https://github.com)")
+		}
```

Comment:
Let's compile these regex's as global vars, this way we can move them to the top of the file and make it easier to find and modify them.